core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1091,9 +1091,10 @@ private static String get(JSONObject o, String prop) {
         public final String minimumJavaVersion;
         /**
          * Categories for grouping plugins, taken from labels assigned to wiki page.
-         * Can be null.
+         * Can be {@code null} if the update center does not support plugin categories.
```

Comment:
This is wrong, update centers (like CloudBees' ) do support categories, they just don't have an empty list when none are defined for a specific plugin.

Reply:
I am fine with this change @daniel-beck . I think that new API would not hurt anyway, especially if we want to keep expanding the label management logic

```suggestion
         * Can be {@code null} if the update center does not return categories.
```